src/net_processing.cpp

```diff
@@ -2088,6 +2088,9 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                       pfrom->nVersion.load(), pfrom->nStartingHeight,
                       pfrom->GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom->addr.ToString()) : ""),
                       pfrom->m_tx_relay == nullptr ? "block-relay" : "full-relay");
+            if (!pfrom->m_tx_relay) {
+                DumpAnchors(connman->m_anchors_db_path, connman->GetBlockRelayNodeAddresses());
```

Comment:
What do you think about waiting some time like 24hours or one block announcement before to qualify node as anchor? That would require at least some good behavior from it, that said and may favor too much well-reliable nodes and decay network-wide connectivity..

nit: you may use scheduler for DumpAnchors like we do for DumpAddresses, that would avoid file syscalls while holding cs_main

Reply:
> nit: you may use scheduler for DumpAnchors like we do for DumpAddresses, that would avoid file syscalls while holding cs_main

> I like the idea of moving this onto the scheduler thread, but not a huge deal given this only happens once per blocks-only connection.

Agree that using `CScheduler.scheduleFromNow()` for `DumpAnchors` is an improvement. Let's leave it for follow ups.